service/quota.py

```diff
@@ -21,112 +111,87 @@ def _get_hard_limits(identity):
+def _set_network_quota(user_quota, identity):
+    network_values = {
+        'port': user_quota.port_count,
+        'floatingip': user_quota.floating_ip_count,
+        # INTENTIONALLY SKIPPED/IGNORED
+        # 'subnet', 'router', 'network',
+        # 'security_group', 'security_group_rules'
+    }
+    username = identity.created_by.username
+    logger.info("Updating network quota for %s to %s"
+                % (username, network_values))
+    driver = get_cached_driver(identity=identity)
+    tenant_id = driver._connection._get_tenant_id()
 
-    return 5-tuple: ((bool) over_quota,
-                     (str) resource_over_quota,
-                     (int) number_requested,
-                     (int) number_used,
-                     (int) number_allowed)
-    """
-    membership = IdentityMembership.objects.get(identity__uuid=identity_uuid,
-                                                member__name=username)
-    user_quota = membership.quota
+    ad = get_account_driver(identity.provider)
+    admin_driver = ad.admin_driver
+    admin_driver._connection._neutron_update_quota(tenant_id, network_values)
+    return
 
-    current = get_current_quota(identity_uuid)
-    logger.debug("Current Quota:%s" % current)
-    cur_cpu = current['cpu']
-    cur_ram = current['ram']
-    cur_disk = current['disk']
-    cur_suspended = current['suspended_count']
-
-    # Add new size to current, check user quota
-    new_cpu = cur_cpu
-    new_ram = cur_ram
-    new_disk = cur_disk
-    if esh_size:
-        new_cpu += esh_size.cpu
-        new_ram += esh_size.ram
-        new_disk += esh_size.disk
-        logger.debug("Quota including size: %s"
-                     % ({'cpu': cur_cpu, 'ram': cur_ram,
-                         'disk': cur_disk}))
-    if resuming:
-        logger.debug("User is resuming an already suspended instance")
-        new_suspended = cur_suspended
-    else:
-        new_suspended = cur_suspended + 1
-        logger.debug("User attempting to suspend/launch another instance")
-
-    # Quota tests here
-    if new_cpu > user_quota.cpu:
-        logger.debug("quota exceeded on cpu: %s"
-                     % user_quota.cpu)
-        return (True, 'cpu', esh_size.cpu, cur_cpu, user_quota.cpu)
-    elif new_ram > user_quota.memory * 1024:  # Quota memory GB -> MB
-        logger.debug("quota exceeded on memory: %s GB"
-                     % user_quota.cpu)
-        return (True, 'ram', esh_size.ram, cur_ram, user_quota.memory)
-    elif not resuming and new_suspended > user_quota.suspended_count:
-        logger.debug("Quota exceed on suspended instances: %s"
-                     % user_quota.suspended_count)
-        return (True, 'suspended instance', 1,
-                cur_suspended, user_quota.suspended_count)
-    return (False, '', 0, 0, 0)
+
+def _set_volume_quota(user_quota, identity):
+    volume_values = {
+        'volumes': user_quota.storage_count,
+        'gigabytes': user_quota.memory,
```

Comment:
Is it wrong that I associate `memory` with `"ram"`?  I assume that this is really "the size of the volume, in gigabytes". Am I right?